### ISSUE TITLE
Not working with Spigot 1.11.2 ?

### ISSUE THREAD

**Issue by smmmadden** [KaiKikuchi/QuickShop#103]
Hi - I have been searching for a replacement of ChestShop and found this one to look like a great contender.  I downloaded the plugin, put it into my plugin folder and restarted the server with having the permissions for players at the world level (all users), and the admin permissions granted just to me in my permission group.  I place a chest down, strike it with a block and the chest opens, but nothing shows in the chat.  I can use any of the /qs commands, but trying to /qs create 5.00 I get a message saying I must be looking at the chest (which I am).  So the messaging appears to be fine, permissions appear to be fine and I've even changed the force-bukkit-chat-handler to true (just in case), but same results.

I removed the ChestShop plugin (jar & folder) in case it was conflicting with this one, but same results.  I am using Vault 1.5.6-b49, EssentialsX 2.01_b465, PEX 1.23.4 and many others.  Server is running under Windows 10 Professional x64.  If you need the list of them, let me know.  Thanks!!! -Steve

p.s.  I also tried adding blocks to the config.yml but even those do not register.
shop-blocks:
    - CHEST
    - TRAPPED_CHEST
    - DIRT
    - LOG
    - COOKED_BEEF

**Comment by KaiKikuchi** [comment]
Sorry I have tested it with Spigot 1.11.2 and it works just fine. Maybe you can send me your server log so I can find more info about your issue?

Also be sure to add Vault and an economy plugin, like SaneEconomy.

**Comment by smmmadden** [comment]
Currently using Fe - could that be it?  Logs all load up the plugins, but I've attached the latest in case I missed something.

Tried attaching a .log and .zip.  neither worked even though it says .zip should.  Had to add it the old fashion way. :-)

EDIT by Kai:
You can use pastebin.com
http://pastebin.com/8GSyHAnK

**Comment by huluhulua** [comment]
I have the same problem. Holding an item left clicks a chest. Nothing happened. No error on console, no error on screen.

I am using latest (2017-01-24) Spigot build git-Spigot-b32c8f8-46226d6 (MC: 1.11.2)
**Edit:**
I think it is conflict with [Machine](https://www.spigotmc.org/resources/machines-automated-drills-turrets-conveyer-belts-and-more.23357/), I remove Machine plugin, and it works again.

**Comment by KaiKikuchi** [comment]
@huluhulua thank you for your report.

@smmmadden 
Yes, it's likely to be another plugin doing bad stuff.

QuickShop is not doing anything weird and it's just using normal Bukkit events.
I suggest you checking the plugins you added and remove anything that may cause these issues to your server.

If you find the plugin causing these issues, please report it here! Thank you.

**Comment by smmmadden** [comment]
Hi KaiKikuchi, several users are reporting the same issue here and on Spigot, but you believe it is another plugin conflicting with this one?  I've removed chestshop from the mix, provided the startup logs you asked for but yet the issue persists.  If you can think of which plugin is preventing QuickShop from working, that would be great. :-)  Closing it as is however, doesn't make the issue go away and I really wanted to use this plugin and promote it on my site and Spigot. May not mean much I know, but it is what I do for other plugins and developers to help each other.

Seriously though, I do have a number of plugins as shown above and they are all playing nicely together - the only change made was remove ChestShop (since it would definitely cause issues), add QuickShop, remove Fe and replaced with your suggestion of SaneEconomy (since this is being supported).  Those are literally the only changes to the world.  I tried left click, right click, shift left and shift right and all do not place the sign or create a shop for the chest.  The wiki doesn't say which one to use, so I'm assuming it was right mouse click like most plugins do.

If it helps, I'm using Spigot 1.11.2 latest build (I keep them current) running on Windows 10 Professional x64 and running with Java 1.8.0_112 x64.

Steps I followed from the home page of Quickshop.  I get to step 2 and click the chest, but nothing appears in the chat window and no sign on the chest.  Are you using a different chat system and perhaps that's why it works for you and not others?  I use EssentialsX2.0.1 build 467 plugin family.
**How to Create a Shop**
1.Place a chest on the ground
2.Hit the chest with the item you want to trade
3.Type in price for the item (As prompted) in chat
4.Fill the chest with the items you wish to sell

**Comment by KaiKikuchi** [comment]
Issues can always be reopened, if needed! Also you can still reply on closed issues.

I say it's another plugin conflict because it works fine on my server, so it's working with a basic spigot test server with QuickShop, Vault and SaneEconomy only.

I am asking what's in your server so I can make a test server with the same stuff so I can do some test.

**Comment by KaiKikuchi** [comment]
Are you using any of the addons for EssentialsX?
![image](https://cloud.githubusercontent.com/assets/8842403/22276098/b4727316-e2b1-11e6-8dd8-7ae9c591858f.png)

**Comment by KaiKikuchi** [comment]
I just tested with QuickShop + Vault + SaneEconomy + EssentialsX 2.0.1... it works fine for me.

You're doing that on survival mode, right?

**Comment by smmmadden** [comment]
Wow, quick reply :-)

Yes, I use all except for XMPP with build 465 thru 467.  I tried using it in Creative & Survival modes.  Here is a full list.
![plugins](https://cloud.githubusercontent.com/assets/10309346/22276560/8567e5da-e282-11e6-9ee7-cc74808cf7c5.png)

**Comment by KaiKikuchi** [comment]
I tried with all but XMPP, QuickShop works fine. You may want to send me your full plugins list

**Comment by smmmadden** [comment]
are you running on Windows 10 Pro x64 or Linux?  Could be an O/S issue, though it would be rare.  I am also not using MySQL.

**Comment by KaiKikuchi** [comment]
I am currently runnning Windows 10 x64 too. I highly doubt an OS issue. I'm not using MySQL either on my test server.

![image](https://cloud.githubusercontent.com/assets/8842403/22276710/cc17d958-e2b5-11e6-918b-359be0716331.png)

**Comment by smmmadden** [comment]
I love a challenge - the only thing that I can think ties to a chest or the sign would be SaneEconomy, Jobs Reborn (paid for blocks), EditableSign which allows a command to edit a placed sign w/o breaking it and LWC.  I've unlocked the chest and locked the chest with no difference.  Guess I'd need to remove plugins one at a time to figure out where it is.  Hoping not to have to remove any of them though.  I'll see if I can narrow it down further - at least we'll know which one is causing it - besides the other one above "Machines".

**Comment by smmmadden** [comment]
I should have done this sooner, but wondering if it is the config.  I'm trying to test this out with COOKED_BEEF and did add it to the config.yml.  Is it possible that the block I'm using won't work?
I'm literally right clicking the chest with cooked_beef in my hand, the chest opens, I press ESC to close it, no message in Chat to put the price.

shop-blocks:
    - CHEST
    - TRAPPED_CHEST
    - DIRT
    - LOG
    - COOKED_BEEF

**Comment by KaiKikuchi** [comment]
You have to click the chest while in survival mode. A message will ask you to type the price in chat. After you do that, a sign will appear on the chest (only if the block in front of the chest is empty). After doing this, clicking the chest will asks you to type in chat how many of the item you want to buy.

**Comment by KaiKikuchi** [comment]
Also I am not sure what you've added to the config. I mean, you don't need to add COOKED_BEEF anywhere on the config. I suggest deleting the QuickShop config and try with the default one.

**Comment by smmmadden** [comment]
I'll try doing that again and let it create the file in case there is an extra character it doesn't like.  I removed all plugins except for Essentials, PEX, PlugMan, QuickShop, SaneEconomy, Vault, WorldBorder and WorldEdit.  Any more and I'll have a Vanilla server again. :-)  Had removed the QuickShop folder completely and restarted.  No go.  Repeated the same after clean restart. Gotta be something I'm doing different than you to get it working.  Is there any step missing from the steps to create it?  It has to be something so trivial that its easy to miss.  :-(  I do appreciate your patience!

**Comment by KaiKikuchi** [comment]
I suggest trying without Plugman. Also it could be that the config got messed up.

**Comment by smmmadden** [comment]
removed QuickShop folder
deleted jar file
downloaded QuickShop jar file from bukkit
removed PlugMan folder & jar
Started server, tested and still no prompting in chat.
I have all the permission nodes in my group for PermissionsEx and tried with the force-bukkit-chat-handler: true

Nothing is written to the qs.log (just an empty log).  Is there a hidden debug option I can turn on to see if the plugin is being called and why it isn't registering the click event?

**Comment by KaiKikuchi** [comment]
yes you can add debug: true at the end of QuickShop's config. Also be sure QuickShop starts correctly by checking server logs. Maybe try with just QuickShop, Vault, and SaneEconomy? If it doesn't work with just that... then I have completely no idea (other than the config being broken somehow).

**Comment by smmmadden** [comment]
debug wasn't making it to the log or the console, but the console does show quickshop loading up with no warnings or errors.  I re-checked the PermissionsEx configuration and while I default to the group called "OwnerAdmin", it appears like some permission nodes are not getting added.  This would definitely explain why nothing works.  Here is what's in my permissions.yml.
    - quickshop.use
    - quickshop.create.sell
    - quickshop.create.buy
    - quickshop.create.double
    - quickshop.create.changeprice
    - quickshop.bypass.ItemID
    - quickshop.find
    - quickshop.unlimited
    - quickshop.setowner
    - quickshop.other.destroy
    - quickshop.other.open
    - quickshop.other.price
    - quickshop.refill
    - quickshop.empty
    - quickshop.clean
I checked PEX and it does show me having access to the list above.   This is leading me to believe, it isn't a matter of permissions or plugin interfering.  It is either the steps I do to create the chest shop or the configuration file that gets created isn't enabling something?  I've run out of options to pin point this to any one area.
Not sure what's up with this editor, I tried to zip up the folder with all the files, but zip's are not allowed even though it says it is? :-)
[config.pdf](https://github.com/KaiKikuchi/QuickShop/files/728784/config.pdf)

**Comment by KaiKikuchi** [comment]
As I said, I would try without anything else but SaneEconomy, Vault and QuickShop. Remove pex etc... try if it finally works, in that case it could be a permission issue?

**Comment by smmmadden** [comment]
I've done just that removing the PEX plugin - you need permissions somewhere and that controls everything.  What' I'd like to know is the **exact steps to get it working** which I eluded to a couple times.  

I removed everything down to the bare necessities so removing everything other than vault, saneeconomy and quickshop would nearly bring me to a vanilla server which is not what I want. :-).  

That's why I'm thinking it has to do with the steps to get it working or permission nodes as I added and posted above are wrong.  Note also, that while I put debug: true at the bottom of the file, nothing wrote into the qs.log which makes it appear like it isn't loading, but server console/logs say it did.

**Comment by KaiKikuchi** [comment]
Sorry debug: true doesn't add any new message to the server load. Anyway, I was asking to try removing everything for a test. I didn't mean you have to keep your server without everything.

Anyway QuickShop doesn't need any particular configuration. You can just download spigot, put it on a folder, add SaneEconomy, Vault and QuickShop to the plugins folder, and it'll work fine. PEX is not required as you can make a shop without any particular permission, and eventually /op gives you all permissions by default.

If you want to make a chest, by default, you just place a chest, then left click with an item the chest directly. It'll ask you in chat the price you want to set for this item. After you send the price in chat, you can put stuff inside the chest and clicking on the chest will tell you how many items do you want to buy.

If you set it to use shift+click instead, then shift+click.

You can also find videos about QuickShop:
- https://www.youtube.com/watch?v=gyMyE7Pz1zs
- https://www.youtube.com/watch?v=eJsv7fqaXNk

**Comment by smmmadden** [comment]
Thanks, the steps you provided did the trick.  The problem is that left click is used to break blocks, especially in creative mode, so never a good choice to use in this example.  Right click only opens a chest unless there is a sign on it already, then it works off of whatever it is.  

I changed the config.yml to "sneak-to-create: true" and "sneak-to-trade: true" and then using the left shift/left click, I got prompted for a price.  So correct me if I'm wrong here, but that sounds like the left click event without sneak (shift), is the problem and not technically another plugin conflict - even though one could argue it is. :-)  Since the default behavior of Minecraft is left click break, right click place or take action on, perhaps keeping the setting by default to be true for these would work better for everyone initially.  Then they can try to switch it off (false) should they do something non-standard?

What do you think?  I haven't begun testing, but I'll be doing that later tonight and get back to you.

Thanks again Kai - Steve

**Comment by KaiKikuchi** [comment]
Well, I've said to use survival mode here: https://github.com/KaiKikuchi/QuickShop/issues/103#issuecomment-275006522

**Comment by smmmadden** [comment]
I know you did and I have been in survival mode right along.  The left click is still used for breaking.  I guess the question is, how long do I have to hold down the left mouse button for it to register and at the same time NOT break the chest? :-)

**Comment by KaiKikuchi** [comment]
Well, you have to hold the click to break a chest... and it requires some time to break it. You just have to left click shortly. Eventually, you can use /qs create while looking at the chest.

**Comment by smmmadden** [comment]
yeah, its the eventually part that concerns me.  Of course now that I tried it with the shift/left click, it works.  I changed the config to reset the sneak to false on both, but shift/left click still worked and was the only way to get the message to enter in the amount.  Don't you believe having users do the shift/left click is better than reporting it doesn't work unless they do?  I understand its been this way and always probably worked, but as Mojang/Microsoft keep changing the games dynamics, I think it is safer and more reliable to shift/left click.   At the end of the day, you want all the users to have a positive experience with any software and to work OOTB.   At least this is what I've driven home to folks for decades. :-)
